src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -857,6 +857,58 @@ public int Length
             }
         }
 
+        private void InternalCopyTo(Span<byte> span, int arrayLength)
```

Comment:
Leave the code duplicated instead? Maybe bring the checks into this method (and make it aggressively inlined)?